scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py

```diff
@@ -16,7 +16,7 @@
 from azureml.assets.util import logger
 
 # Handles {{latest-pypi-version}} and {{latest-pypi-version:flags}}
-LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+)([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")
+LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+?)(?:\[.*\])?([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")

```

Comment:
There wouldn't ever be a need to match `[]`
```suggestion
LATEST_PYPI_VERSION = re.compile(r"([^\"'\s]+?)(?:\[.+\])?([=~]=)\{\{latest-pypi-version(?::(.+))?\}\}")
```